src/lib.rs

```diff
@@ -192,6 +191,18 @@ unsafe extern "C" fn reset_handler() -> ! {
     ::r0::zero_bss(&mut _sbss, &mut _ebss);
     ::r0::init_data(&mut _sdata, &mut _edata, &_sidata);
 
+    match () {
+        #[cfg(has_fpu)]
+        () => {
+            // NOTE(safe) no exception / interrupt that also accesses the FPU
+            // can occur here
+            let scb = &*cortex_m::peripheral::SCB.get();
+            scb.enable_fpu();
```

Comment:
No. That's useless. main **must** not be inlined or whichever function it's inlined into will try to save FPU registers and crash before it can do anything (such as enable the FPU).